File: hanoi.pas
program ideone;
procedure hanoi(n:integer; x:string; y:string; z:string);
begin
	if n>1 then hanoi(n-1,x,z,y);
	writeln(x+'=>'+y);
	if n>1 then	hanoi(n-1,z,y,x);
end;

var n:integer;
begin
	read(n);
	hanoi(n,'a','b','c');
end.

File: hanoi.f90
program hanoi
  implicit none
  integer :: n
  read *, n
  call tower(n, 'a', 'b', 'c')
  stop
contains
  recursive subroutine tower(n, a, b, c)
    implicit none
    integer :: n
    character :: a, b, c
    if (n == 0) then
      return
    end if
    call tower(n - 1, a, c, b)
    print *, a, " => ", b
    call tower(n - 1, c, b, a)
    return
  end subroutine tower
end program hanoi
